doc/en/reference/reference.rst

```diff
@@ -1120,7 +1120,11 @@ Environment variables that can be used to change pytest's behavior.
 
 .. envvar:: CI
 
-When set (regardless of value), pytest acknowledges that is running in a CI process. Alternative to ``BUILD_NUMBER`` variable.
+When set (regardless of value), pytest acknowledges that is running in a CI process:
```

Comment:
It would probably be best for us to create a new docs session somewhere detailing how pytest detects CI and the consequences of that, and link to that session here.

Reply:
Took a cursory look and did not find a suitable place to add this information to, perhaps we can just create a new document in `doc/en/explanation/ci.rst`. This could explain the rationale for auto-detecting CI, mention the two variables, and what pytest does differently when running on CI.

Reply:
I feel like "how we detect we are in a CI env?" and "what are the consequences?" are too short and simple to have a doc session (which I guess is a dedicated file?) somewhere

Reply:
I disagree, I don't think it is a problem having a short document: it is a place to explain this rationale, and to grow in case we change other behaviors in the future depending if we are on CI or not.

Reply:
fair enough! I'll try to dig out the original issue to create a good documentation :)

Reply:
Awesome, thanks! Really appreciate it. 

I think a short document with:

* Rationale
* How CI is detected
* Effects on CI

With one or two paragraphs each, would be perfect.